app/controllers/actions/DataRetrievalAction.scala

```diff
@@ -18,23 +18,28 @@ package controllers.actions
 
 import models.requests.{MovementRequest, OptionalDataRequest}
 import play.api.mvc.ActionTransformer
-import services.UserAnswersService
+import services.{GetTraderKnownFactsService, UserAnswersService}
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.http.HeaderCarrierConverter
 
 import javax.inject.Inject
 import scala.concurrent.{ExecutionContext, Future}
 
-class DataRetrievalActionImpl @Inject()(val userAnswersService: UserAnswersService)
+class DataRetrievalActionImpl @Inject()(val userAnswersService: UserAnswersService,
+                                        val traderKnownFactsService: GetTraderKnownFactsService)
                                        (implicit val executionContext: ExecutionContext) extends DataRetrievalAction {
 
   override protected def transform[A](request: MovementRequest[A]): Future[OptionalDataRequest[A]] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequestAndSession(request, request.session)
 
-    userAnswersService.get(request.ern, request.arc).map {
-      OptionalDataRequest(request, _)
+    for {
+      userAnswers <- userAnswersService.get(request.ern, request.arc)
+      traderKnownFacts <- traderKnownFactsService.getTraderKnownFacts(request.ern)
+    } yield {
+      OptionalDataRequest(request, userAnswers, traderKnownFacts)
```

Comment:
[Question] In what case would trader known facts be optional for this data request and mandatory for DataRequest.

Reply:
I'll raise a tech debt ticket to condense DataRetrievalAction and DataRequiredAction into one across all our repos, so that we don't need the OptionalDataRequest at all. I agree how it's currently built is a little redundant/unnecessarily complicated, but this is a change that would need to be made everywhere so best to do it all in one ticket I think.